Use argparse within chipsec_main

BrentHoltsclaw opened the pull request.

ErikBjorge commented on chipsec/testcase.py:
```diff
@@ -36,6 +36,24 @@ class ExitCode:
     INFORMATION   = 64
     NOTAPPLICABLE = 128
 
+    Help_String = """\
```
Can we name this help_epilog to be a little more clear on how it is used.

ErikBjorge commented on chipsec_main.py:
```diff
@@ -373,117 +356,49 @@ def run_all_modules(self):
-                self._deltas_file = a
-            else:
-                assert False, "unknown option"
+        parser = argparse.ArgumentParser(prog='chipsec_main.py',usage='%(prog)s [options]',formatter_class=argparse.RawDescriptionHelpFormatter,epilog=ExitCode.Help_String,add_help=False)
```
Do we want to leave the .py off the program name.  This may be better for people that install it on a test image.

BrentHoltsclaw commented:
@ErikBjorge thanks for the suggestions.  I made the fixes and squashed them.  

I also changed the ordering so that we do not try to do anything with the chipset or helpers before parsing arguments.
